tpp-app/src/services/LogSymptomsService.js

```diff
@@ -0,0 +1,69 @@
+import AsyncStorage from '@react-native-async-storage/async-storage';
+
+import {FLOW_LEVEL} from './utils/constants'
+import { initializeEmptyYear, GETsymptomsForDate, isValidDate } from './utils/helpers'
+
+
+/**
+ * Posts the user's symptom data into the local storage for the given date.
+ * @param day number (1st day of month = 1)
+ * @param month number (January = 1)
+ * @param year number
+ * @param symptoms Symptoms object
+ */
+export const POSTsymptomsForDate = async (day, month, year, symptoms) => new Promise( (resolve, reject) => {
+    // Check that symptoms object is not all null or not empty
+    let notEmpty = Object.values(symptoms).some((symptom) => symptom !== null)
+    if (!symptoms || notEmpty) {
+        reject("No symptoms to record");
+    }
+
+    // Check that date, month, year combo is valid
+    if (!isValidDate(day, month, year)) {
+        reject("Not a valid date");
+    }
+
+    // Try to POST new symptoms in async storage
+    try {
+        // Get the year's data or set to empty year
+        const yearData = JSON.parse(await AsyncStorage.getItem(year.toString())) ?? initializeEmptyYear(year);
+
+        yearData[month-1][day-1] = symptoms
+
+        // post symptoms to storage
+        await AsyncStorage.setItem(year, JSON.stringify(yearData))
+            .then(() => resolve())
+            .catch((e) => {
+                console.log(JSON.stringify(e));
+                reject(`Unable to mergeItem and post symptoms for this day, month, year: ${day, month, year}`);
+            });
+    } catch (e) {
+        console.log(`POSTsymptomsForDate error: ${JSON.stringify(e)}`);
+        reject("Something went wrong");
+    }
+})
+
+export const LogMultipleDaysService = {
+    LogMultipleDayPeriod: async (dates) => {
+        // run this code for each value in the dates array
+        dates.map(async (date)=>{
+            const year = date.year;
+            const month = date.month;
+            const day = date.day;
+
+            let symptoms = await GETsymptomsForDate(day, month, year)
+
+            if (symptoms.flow == null){
```

Comment:
suggest 
```if (symptoms.flow == null || symptoms.flow == FLOW_LEVEL.NONE) {```
because flow=null and flow=NONE have same meaning, the user says there's no period. But if they select it in the log multiple days screen, they want to set the non-period day to be a period day so it should be set to FLOW_LEVEL.MEDIUM

Reply:
True true! all good now.